pkg/controller/baremetalhost/baremetalhost_controller.go

```diff
@@ -286,7 +333,26 @@ func (r *ReconcileBareMetalHost) Reconcile(request reconcile.Request) (result re
 			"old", host.Status.Provisioning.State,
 			"new", actionName,
 		)
+
 		host.Status.Provisioning.State = actionName
+
+		// Update the start time for the operational phase we are in.
+		timestamp := metav1.Now()
+		switch actionName {
+		case metal3v1alpha1.StateRegistering:
+			host.Status.OperationHistory.Register.Start = timestamp
+			host.Status.OperationHistory.Register.End = metav1.Time{}
+		case metal3v1alpha1.StateInspecting:
+			host.Status.OperationHistory.Inspect.Start = timestamp
+			host.Status.OperationHistory.Inspect.End = metav1.Time{}
+		case metal3v1alpha1.StateProvisioning:
+			host.Status.OperationHistory.Provision.Start = timestamp
+			host.Status.OperationHistory.Provision.End = metav1.Time{}
+		case metal3v1alpha1.StateDeprovisioning:
+			host.Status.OperationHistory.Deprovision.Start = timestamp
+			host.Status.OperationHistory.Deprovision.End = metav1.Time{}
+		}
```

Comment:
You could avoid some of the duplication by doing e.g.

```
var metric *OperationMetric
switch actionName {
    case StateRegistering:
        metric = &host.Status.OperationHistory.Register
    case ...
}
if metric != nil {
    metric.Start = metav1.Now()
    metric.End = metav1.Time{}
}
```

(You could avoid *all* of the duplication by doing introspection, but that's definitely overkill :D)

Splitting the first part out into a function would make it reusable, which might be helpful to also record the end of each state in a single place (that could happen in `updateHostStateFrom()` once this is rebased).